basyx.components/basyx.components.docker/basyx.components.AASServer/src/main/java/org/eclipse/basyx/components/aas/mongodb/MongoDBAASAggregator.java

```diff
@@ -417,31 +419,18 @@ public void setRegistry(IAASRegistry registry) {
 	 * @deprecated This method is used with the old, deprecated Constructors. Use
 	 *             {@link MongoDBAASServerComponentFactory} instead
 	 */
-	@SuppressWarnings("deprecation")
 	@Deprecated
 	public void setConfiguration(BaSyxMongoDBConfiguration config) {
-		// set mongoDB configuration
-		this.config = config;
 		MongoClient client = MongoClients.create(config.getConnectionUrl());
-		this.mongoOps = new MongoTemplate(client, config.getDatabase());
-		this.aasCollection = config.getAASCollection();
-		this.smCollection = config.getSubmodelCollection();
+		setMongoDBConfiguration(config, client);
 
-		// Create API factories with the given configuration
-		this.aasApiProvider = aas -> {
-			MongoDBAASAPI api = new MongoDBAASAPI(config, aas.getIdentification().getId());
```

Comment:
If you give here and in line 437 also the client to the MongoDB*API-Constructors will also reduce new client creations.